lib/modules/manager/buildpacks/__fixtures__/project.toml

```diff
@@ -0,0 +1,35 @@
+[_]
+schema-version = "0.2"
+
+[io.buildpacks]
+builder = "registry.corp/builder/noble:1.1.1"
+
+[[io.buildpacks.group]]
+uri = "docker://buildpacks/java:2.2.2"
+
+[[io.buildpacks.group]]
+uri = "buildpacks/nodejs:3.3.3"
```

Comment:
I see now, this is a docker image, but our docker datasource will never find it, because it needs to be translated via build packs registry to the real image.

https://registry.buildpacks.io//api/v1/buildpacks/heroku/nodejs/3.2.11

so I'm pretty sure we need the buildpacks datasource first.

Reply:
We'll start working on that soon.

Reply:
> I see now, this is a docker image, but our docker datasource will never find it, because it needs to be translated via build packs registry to the real image.
> 
> https://registry.buildpacks.io//api/v1/buildpacks/heroku/nodejs/3.2.11
> 
> so I'm pretty sure we need the buildpacks datasource first.

No, this example is perfectly valid and will work with the current docker datasource. At the moment, there are several different ways to add a buildpack to the `project.toml` file: by `id` or `uri`. When the buildpack URI is specified and it does not fall under the cases listed [here](https://github.com/sap-contributions/renovate/blob/1b85bedf5f576afde6d2b9acc273b9ac5dae04ea/lib/modules/manager/buildpacks/extract.ts#L31-L35), the URI is considered to be a direct OCI image reference, so no buildpack registry lookup should be performed.

The URIs which must be looked up against the buildpacks registry will be ignored and only direct references are picked up by the manager. We wanted to start working on the buildpacks manager incrementally, first by using already existing datasource for updating docker images. We need a separate `buildpacks registry` (or similar) datasource only when the manager must support the following cases:

- `id` + `version` are set instead of the `uri`
- `uri` starts with `urn:cnb:registry`
- `uri` matches the following regex: https://github.com/buildpacks/pack/blob/3a22a7ffef9e69b33ba32fcbfbe6491a09d5397c/pkg/buildpack/locator_type.go#L39